proxy/src/main/java/com/wavefront/agent/queueing/QueueController.java

```diff
@@ -51,10 +47,10 @@
   @SuppressWarnings("UnstableApiUsage")
   protected final RateLimiter reportRateLimiter = RateLimiter.create(0.1);
 
-  private AtomicLong currentWeight = null;
-  private final AtomicInteger queueSize = new AtomicInteger();
+  private long currentWeight;
+  private int queueSize;
```

Comment:
do we think it's safe to remove the atomic?

Reply:
Yes, we create just one instance of this class for each "port", they are private members, update only on thread.
I don't see the need for Atomic operations.

Reply:
but can be put Atomic back if you think is best

Reply:
The metric reporting runs in a different thread, actually.  Although that thread does is read the int.  So if only one thread is writing we are still ok.  I'm fine keeping it int.